includes/shortcodes.php

```diff
@@ -31,10 +31,16 @@ function fastwc_add_shortcodes() {
  * @return string
  */
 function fastwc_shortcode_product_button( $atts ) {
-	return fastwc_shortcode_button_template(
-		'buttons/fast-checkout-button',
-		$atts
-	);
+	if ( fastwc_should_hide_checkout_button() ) {
```

Comment:
FWIW, this function does rely on two global variables that are set in the `wc_product_addon_start` and `woocommerce_grouped_product_list_before` hooks. Depending on where they try to jam this shortcode, those variables might not be set. It also relies on the global `$product` variable.
I think it's probably fine to keep this as is, but it's something we should keep in the back of our heads in case it comes up in a corner case.

Reply:
@brikr I'm revisiting this. Do you know why we use those globals to check for addons and grouped products? There is another check for addons in the cart button check, and I'm wondering why we don't just use that to check if the product has addons - https://github.com/fast-af/fast-checkout-woocommerce/blob/main/includes/checkout.php#L158-L167.

I'm sure we can find a similar way to check for grouped products as well, and just remove the dependency on the hooks and global variables. What do you think?